### PR TITLE
new(ci): added gha.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

**What this PR does / why we need it**:

Ported CI to gha.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by FedeDP** [comment]
I think we might want to keep circleci in, and the open a second PR where i delete it; this way, we can test that everything works before ditching it.

**Comment by FedeDP** [comment]
Dropped the circleci cleanup.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
```
Comment (leogr) [inline .github/workflows/release.yaml:6]: We need to support [pre-releases](https://semver.org/#spec-item-9)

Comment (FedeDP) [reply .github/workflows/release.yaml:6]: I think we can post-pone this until we actually need it, to be consistent across the falcosecurity.

Comment (leogr) [reply .github/workflows/release.yaml:6]: agree

**Review by leogr — APPROVED** [review]

**Pull request merged**